module-bsp/board/rt1051/bellpx/bsp/switches/switches.cpp

```diff
@@ -48,11 +48,10 @@ namespace bsp::bell_switches
 
     struct DebounceTimerState
     {
-        const DebounceTimerId id{DebounceTimerId::Invalid};
-        const NotificationSource notificationSource{NotificationSource::Invalid};
+        const KeyId id{KeyId::Invalid};
         const std::shared_ptr<drivers::DriverGPIO> gpio;
         const BoardDefinitions pin{-1};
-        KeyEvents lastState{KeyEvents::Pressed};
+        volatile KeyEvents lastState{KeyEvents::Pressed};
```

Comment:
is it intended? `volatile`

Reply:
Might work without it, but that variable is being modified from IRQ so it is safer to prevent the compiler from optimizing it

Reply:
can we resolve this conversation?

Reply:
Resolving